tests/suites/target_test.function

```diff
@@ -354,6 +354,8 @@ int execute_tests( int args, const char ** argv )
         data = receive_data( &data_len );
         if ( data == NULL )
             continue;
+        if ( data[0] == NULL && data_len == 1 )
```

Comment:
Would recommend using `NULL` for data pointers and `'\0'` for zero-valued characters. So, `'\0'` here.

Reply:
If it's important to call `mbedtls_platform_terminate()`, is it also important to call `free( data );` to avoid a memory leak between test runs? Do we not reboot the target between tests? If we rebooted between tests, wouldn't we not need to call free() or mbedtls_platform_terminate()?

Reply:
I believe that the device is rebooted between tests, but I have noticed in a target, that once I get a failure, then all other tests fail. Since mbedtls_platform_terminate()` is currently responsible for clearing the hw crypto engine, perhaps sometimes a reboot is only a soft reboot, clearing memory, but not clearing interrupts \ hw.